README.md

```diff
@@ -45,102 +53,241 @@ console.log(result);
 nextOperation(result)
 // ... danger
 ```
-Here our **addTen** function has unknowingly worked with two different data types, and our lovely JavaScript engine has coerced them into strings and concatenated them together to produce **[object Object]10**, which is not helpful.
+Here our `addTen` function has unknowingly worked with two different data types: a `Number` and an `Object`. When asked to add a number to an object, our lovely JavaScript engine coerces them both to strings and concatenates them together to produce `'[object Object]10'`, which is not helpful.
+
+If `arrayOfNumbersIThink` was retrieved from an API call or user input, we can't always be certain the values will be what we expect. How can we deal with these situations?
+
+## Kinds Of Errors
+Broadly speaking, errors come in two kinds [[8]](#resources):
+* **Programmer errors**: These are _bugs_; they are unintended and/or unanticipated behaviour of the code, and they can only be fixed by changing the code (e.g. calling a function with the wrong number of arguments)
+* **Operational errors**: These are runtime errors that are usually caused by some external factor (e.g. any kind of network error, failure to read a file, running out of memory, etc.)
+
+How you should handle any given error depends on what kind of error it is. Operational errors are a normal part of the issues a program must deal with. They typically should not cause the program to terminate or behave unexpectedly. By contrast, programmer errors are by definition unanticipated, and may potentially leave the application with unpredictable state and behaviour. In this case it is usually best to terminate the program.
+
+There is, however, no blanket rule for what to do; each error represents a specific problem in the context of an entire application and the appropriate response to it will be heavily context dependent.
+
+## Approaches
+Good error handling is typically not something that can just be bolted onto an existing program as an afterthought. Well conceived error handling will affect the structure of the code. In JavaScript and Node.js, there are a number of approaches, some of which are explored below.
+
+### General Principles
+Regardless of the chosen approach, there are some principles which can be generally applied:
+1. **Be consistent, not ad-hoc.**
+  * Inconsistent approaches to error handling will complicate your code and make it much harder to reason about.
+2. **Try to trip into a failure code path as early as possible.**
```

Comment:
Can we explain what failure code path means? Or add a short example of _how_ to trip into a failure code path as early as possible?